src/lib.rs

```diff
@@ -283,39 +282,38 @@ impl Keccak {
         }
 
         // Xor in the last block
-        xorin(&mut a[self.offset..], &input[ip..], l);
+        let offset = self.offset;
+        xorin(&mut self.a_mut_bytes()[offset..], &input[ip..], l);
         self.offset += l;
     }
 
     pub fn pad(&mut self) {
-        let mut a = as_mut_bytes_slice(&mut self.a);
-
         let offset = self.offset;
         let rate = self.rate;
 
+        let delim = self.delim;
         unsafe {
-            *a.get_unchecked_mut(offset) ^= self.delim;
-            *a.get_unchecked_mut(rate - 1) ^= 0x80;
+            let aa = self.a_mut_bytes();
+            *aa.get_unchecked_mut(offset) ^= delim;
+            *aa.get_unchecked_mut(rate - 1) ^= 0x80;
         }
     }
 
     // squeeze output
     pub fn squeeze(&mut self, output: &mut [u8]) {
-        let a = as_bytes_slice(&mut self.a);
-
         let outlen = output.len();
         let rate = self.rate;
 
         // second foldp
         let mut op = 0;
         let mut l = outlen;
         while l >= rate {
-            setout(&a, &mut output[op..], rate);
+            setout(self.a_bytes(), &mut output[op..], rate);
```

Comment:
this is actually a pretty important change; I didn't realize it before but `self.a` is aliased here in the prior version